### PR TITLE
test: Modernize and convert some test in `main.js` to async/await

### PR DESCRIPTION
**Pull request by jimmywarting**
## Purpose
to tidy up and get rid of some indentation hell 

## Changes
No other test file than main.js file was changed.
the test are still the same...

### CODE REVIEW

File: test/main.js [hunk 1]
```diff
@@ -163,237 +161,199 @@ describe('node-fetch', () => {
...
-			headers: {
-				HoSt: 'example.com'
-			}
-		};
-		return fetch(url, options).then(res => {
-			return res.json();
-		}).then(res => {
-			expect(res.headers.host).to.equal('example.com');
-		});
-	});
```
Comment (LinusU) [inline test/main.js:252]: Why was this removed?

File: test/main.js [hunk 2]
```diff
@@ -697,112 +647,102 @@ describe('node-fetch', () => {
...
-		return fetch(url).then(res => {
-			expect(res.status).to.equal(200);
-			expect(res.ok).to.be.true;
-			return expect(res.text()).to.eventually.be.rejectedWith(Error)
-				.and.have.property('message').matches(/Premature close|The operation was aborted|aborted/);
-		});
+		const res = await fetch(url);
+		expect(res.status).to.equal(200);
+		expect(res.ok).to.be.true;
+		return expect(res.text()).to.eventually.be.rejectedWith(Error)
```
Comment (LinusU) [inline test/main.js:655]:
```suggestion
		await expect(res.text()).to.eventually.be.rejectedWith(Error)
```

**Review by LinusU — APPROVED** [review]
Looks good, just a few nits 👍

File: test/main.js (hunk 1 above)
Comment (jimmywarting) [reply test/main.js:252]: we have two identical tests: https://github.com/node-fetch/node-fetch/blob/1028f8370844060ec81279a9b956b0c22d611cf3/test/main.js#L226-L252

Comment (LinusU) [reply test/main.js:252]: One was setting the header as `hOsT`, and tested that it was normalised into `Host`

Comment (jimmywarting) [reply test/main.js:252]: Oh... hmm

Comment (jimmywarting) [reply test/main.js:252]: guess i can take it back

File: test/main.js [hunk 3]
```diff
@@ -222,6 +222,18 @@ describe('node-fetch', () => {
 		const json = await res.json();
 		expect(json.headers.host).to.equal('example.com');
 	});
+
```
Comment (LinusU) [inline test/main.js:225]:
```suggestion

```

**Review by LinusU — APPROVED** [review]
Neat! 👍

**Comment by zelma83** [comment]
> 
> 
> ## Purpose
> to tidy up and get rid of some indentation hell 
> 
> ## Changes
> No other test file than main.js file was changed.
> the test are still the same...
>
